src/nsrt_learning.py

```diff
@@ -76,12 +76,12 @@ def segment_trajectory(trajectory: GroundAtomTrajectory) -> List[Segment]:
     If options are available, also use them to segment.
     """
     segments = []
-    states, actions, all_atoms = trajectory
-    assert len(states) == len(actions) + 1 == len(all_atoms)
-    current_segment_traj : StateActionTrajectory = ([], [])
-    for t in range(len(actions)):
-        current_segment_traj[0].append(states[t])
-        current_segment_traj[1].append(actions[t])
+    traj, all_atoms = trajectory
+    assert len(traj.states) == len(all_atoms)
+    current_segment_traj : Tuple[List[State], List[Action]] = ([], [])
```

Comment:
can you change this so that it's now two lists, `current_segment_states` and `current_segment_actions`?

Reply:
yes, good idea!